programs/squads_multisig_program/src/instructions/proposal_vote.rs

```diff
@@ -119,6 +146,98 @@ impl ProposalVote<'_> {
     }
 }
 
+impl ProposalVoteV2<'_> {
+    fn validate(&self, vote: Vote) -> Result<()> {
+        let Self {
+            multisig,
+            proposal,
+            member,
+            ..
+        } = self;
+
+        // member
+        require!(
+            multisig.is_member(member.key()).is_some(),
+            MultisigError::NotAMember
+        );
+        require!(
+            multisig.member_has_permission(member.key(), Permission::Vote),
+            MultisigError::Unauthorized
+        );
+
+        // proposal
+        match vote {
+            Vote::Approve | Vote::Reject => {
```

Comment:
For example this: i don't think we need any of these, and can do a 
```_ => panic/error``` 
for non cancel actions.